t4_devkit/sanity/format/base.py

```
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from returns.maybe import Maybe, Nothing, Some
+from returns.pipeline import is_successful
+from returns.result import Result, safe
+
+from t4_devkit.schema import SCHEMAS, SchemaBase, SchemaName
+
+from ..checker import Checker
+from ..result import Reason
+from ..safety import load_json_safe
+
+if TYPE_CHECKING:
+    from ..context import SanityContext
+
+
+class FieldTypeChecker(Checker):
+    """Base class for format checkers."""
+
+    schema: SchemaName
+
+    def can_skip(self, context: SanityContext) -> Maybe[Reason]:
+        match context.to_schema_file(self.schema):
+            case Some(x):
+                if not x.exists() and not self.schema.is_optional():
+                    return Maybe.from_value(Reason(f"No '{self.schema.filename}' found"))
```

[question]
I might not have understood it correctly, but if the table is not optional and it does not exist, shouldn't it be considered a failure instead of a skip case?